### ISSUE TITLE
[ACE-Step Epic] Local ACE-Step 1.5 Python Execution Script

### ISSUE THREAD

**Issue by mgcronin** [openzigs/openzigs#337]
## Sub-Issue 2: Local ACE-Step 1.5 Python Execution Script

**Parent Epic:** #335

### Goal
Create the local Python execution layer for ACE-Step 1.5 that runs on the same Mac as the OpenZigs Node.js server, using the Apple Silicon optimized MPS/Metal + MLX implementation.

### Reference Implementation
- **Fork**: [`clockworksquirrel/ace-step-apple-silicon`](https://github.com/clockworksquirrel/ace-step-apple-silicon)
- **Upstream**: [`ACE-Step/ACE-Step-1.5`](https://github.com/ACE-Step/ACE-Step-1.5)

#### 1. Python Generation Script (`src/audio/acestep-generator.py`)
Create a standalone Python script that:
- Accepts JSON input via stdin or command-line arguments
- Loads the Apple Silicon optimized ACE-Step 1.5 pipeline
- Generates music using the DiT + LM pipeline on MPS/Metal
- Outputs the generated audio file path and metadata as JSON to stdout
- Strictly manages memory: `torch.mps.empty_cache()`, `gc.collect()`, and `mx.metal.clear_cache()` (if MLX available) after every generation

#### 2. Input Parameters (JSON)
```json
{
  "tags": "upbeat electronic dance music, 120 bpm, energetic synths",
  "lyrics": "[verse]\nLost in the rhythm of the night\n[chorus]\nDance until the morning light",
  "duration": 30,
  "instrumental": false,
  "ditModel": "acestep-v15-turbo",
  "lmModel": "acestep-5Hz-lm-0.6B",
  "seed": -1,
  "bpm": null,
  "key": null,
  "timeSignature": null,
  "outputDir": "~/.openzigs/gallery",
  "audioFormat": "mp3"
}
```

#### 3. Output (JSON to stdout)
```json
{
  "success": true,
  "audioPath": "/Users/user/.openzigs/gallery/acestep_1709123456789.mp3",
  "duration": 30.0,
  "seed": 42,
  "model": "acestep-v15-turbo",
  "generationTime": 45.2,
  "metadata": {
    "bpm": 120,
    "key": "C Major",
    "timeSignature": "4/4"
  }
}
```

#### 4. Memory Management (Critical)
Since this runs locally alongside the Node.js server:
```python
# After every generation
import torch
import gc

torch.mps.empty_cache()
torch.mps.synchronize()
gc.collect()

try:
    import mlx.core as mx
    mx.metal.clear_cache()
except ImportError:
    pass
```

#### 5. Installation / Setup Script
- Add an install helper that clones and sets up the ACE-Step Apple Silicon fork
- Use `uv` for Python environment management (recommended by the fork)
- Auto-download models on first run (~5 GB)
- Document Python 3.11.x requirement

#### 6. Node.js Wrapper (`src/audio/acestep-service.ts`)
- TypeScript service that spawns the Python script as a child process
- Parses JSON output from stdout
- Handles errors from stderr
- Implements timeout handling (generation can take 45s-3min depending on model/duration)
- Reports progress events via EventEmitter for queue status updates

### Key Technical Constraints
- **Python 3.11.x** required (pinned in `pyproject.toml`)
- **PyTorch 2.4+** for MPS bfloat16 support
- **MLX** is optional but recommended for faster LM inference
- **Backend**: Must use `--backend pt` on macOS (vLLM requires CUDA)
- **Device**: `--device auto` will auto-detect MPS on Apple Silicon

### Performance Expectations (Apple Silicon)
| Scenario | M2 Pro | M3 Max |
|---|---|---|
| 30s song (turbo, 8 steps) | ~45s | ~25s |
| 30s song (SFT, 32 steps) | ~3 min | ~1.5 min |
| LM reasoning (0.6B) | ~10s | ~5s |

### Files to Create
- `src/audio/acestep-generator.py` — Main Python generation script
- `src/audio/acestep-service.ts` — Node.js wrapper/child process manager
- `scripts/setup-acestep.sh` — Installation helper script

### Acceptance Criteria
- [ ] Python script generates music from JSON input
- [ ] Memory is cleaned after every generation
- [ ] Node.js wrapper spawns child process and parses output
- [ ] Error handling for timeouts, Python errors, missing models
- [ ] First-run model download works automatically
- [ ] Seed produces reproducible output
